### PR TITLE
Fix wrong interface name with auto_vlan=yes

### PR DESCRIPTION
**Pull request by mvidner**
Porting #38 to SLE-12

The actual patch has a couple of lines, the rest of the PR makes its surroundings a bit more readable.

### CODE REVIEW

File: src/modules/FcoeClient.rb [hunk 1]
```diff
@@ -1223,107 +1223,33 @@ def WriteSysconfigFiles
...
-              Ops.add(
-                Ops.add(
-                  path(".network.value"),
-                  Ops.get_string(card, "dev_name", "")
-                ),
-                "STARTMODE"
-              ),
-              "nfsroot"
-            )
+            SCR.Write(ifcfg_path + "STARTMODE", "nfsroot")
```
Comment (mchf) [inline src/modules/FcoeClient.rb]: this can be picked out of if-else - shared between both branches

File: src/modules/FcoeClient.rb [hunk 2]
```diff
@@ -1223,107 +1223,33 @@ def WriteSysconfigFiles
...
-                  path(".network.value"),
-                  Ops.get_string(card, "dev_name", "")
-                ),
-                "NAME"
-              ),
-              Ops.get_string(card, "device", "")
-            )
+            SCR.Write(ifcfg_path + "BOOTPROTO", "static")
+            SCR.Write(ifcfg_path + "STARTMODE", "nfsroot")
+            SCR.Write(ifcfg_path + "NAME", Ops.get_string(card, "device", ""))
```
Comment (mchf) [inline src/modules/FcoeClient.rb]: why not to get even rid of last remaining ```Ops.get_string(...)```?

File: src/include/fcoe-client/complex.rb [hunk 3]
```diff
@@ -447,11 +435,7 @@ def HandleInterfacesDialog(id, event)
             # if /etc/sysconfig/network/ifcfg-<if>.<vlan> already exists
             # call 'ifup' for the interface (creates /proc/net/vlan/<if>.<vlan>)
             if FileUtils.Exists(ifcfg_file)
-              cmd_ifup = Builtins.sformat(
-                "ifup %1.%2",
-                dev_name,
-                Ops.get_string(card, "vlan_interface", "")
-              )
+              cmd_ifup = Builtins.sformat("ifup %1.%2", dev_name, vlan_interface)
```
Comment (mchf) [inline src/include/fcoe-client/complex.rb]: we have ```SetIfaceUp``` in yast2-network ;-)

File: src/include/fcoe-client/complex.rb [hunk 4]
```diff
@@ -392,7 +392,11 @@ def HandleInterfacesDialog(id, event)
           )
         end
 
-        command = Builtins.sformat("fipvlan -c -s %1", dev_name)
+        if card["auto_vlan"] == "yes" || vlan_interface == "0"
+          command = Builtins.sformat("fipvlan -c -s -f '-fcoe' %1", dev_name)
+        else
+          command = Builtins.sformat("fipvlan -c -s %1", dev_name)
```
Comment (mchf) [inline src/include/fcoe-client/complex.rb]: You did a nice cleanup, so you can go even further: ```"fipvlan -c -s #{ dev_name}"```

File: src/include/fcoe-client/complex.rb [hunk 5]
```diff
@@ -774,9 +759,7 @@ def HandleInterfacesDialog(id, event)
       nil
     end
 
-    def HandleConfigurationDialog(id, event)
-      event = deep_copy(event)
-      action = Ops.get(event, "ID")
+    def HandleConfigurationDialog(_id, _event)
```
Comment (mchf) [inline src/include/fcoe-client/complex.rb]: what is the reason why this cannot be removed completely?

File: src/modules/FcoeClient.rb [hunk 6]
```diff
@@ -1223,107 +1222,34 @@ def WriteSysconfigFiles
       success = true
 
       Builtins.foreach(netcards) do |card|
-        if Ops.get_string(card, "fcoe_vlan", "") != @NOT_AVAILABLE &&
-            Ops.get_string(card, "fcoe_vlan", "") != @NOT_CONFIGURED
+        fcoe_vlan = card.fetch("fcoe_vlan", "")
+        if fcoe_vlan != @NOT_AVAILABLE &&
+            fcoe_vlan != @NOT_CONFIGURED
```
Comment (mchf) [inline src/modules/FcoeClient.rb]: alternative which I sometimes prefer: ```![@NOT_AVAILABLE, @NOT_CONFIGURED].include? fcoe_vlan```

File: src/include/fcoe-client/complex.rb (hunk 5 above)
Comment (mvidner) [reply src/include/fcoe-client/complex.rb]: We could, but I just have to stop the refactoring somewhere.

File: src/include/fcoe-client/complex.rb (hunk 3 above)
Comment (mvidner) [reply src/include/fcoe-client/complex.rb]: Having learned about it, well, it is such a lone function (nobody uses it!) that I don't feel comfortable using it.
BTW it does not document its return value: http://www.rubydoc.info/github/yast/yast-network/Yast%2FNetworkRoutinesInclude%3ASetIfaceUp

Comment (mchf) [reply src/include/fcoe-client/complex.rb]: well it used to be used. Currently it is replaced by something more friendly to wicked.

File: src/include/fcoe-client/complex.rb (hunk 5 above)
Comment (mchf) [reply src/include/fcoe-client/complex.rb]: you're touching it anyway and dropping unused method makes more sense to me over adding '_' to satisfy a convention ;-)

Comment (mvidner) [reply src/include/fcoe-client/complex.rb]: An *empty* method is not an *unused* method. You have missed my point.

**Pull request merged**